admin/spec/service/pin-generation.service.spec.js

```diff
@@ -220,4 +221,37 @@ describe('pin-generation.service', () => {
       })
     })
   })
+
+  describe('generateCryptoRandomNumber', () => {
+    it('should generate a random number in specific range', () => {
+      const number = pinGenerationService.generateCryptoRandomNumber(1, 6)
+      expect(typeof number).toBe('number')
+      expect(number >= 0 || number <= 6).toBeTruthy()
+    })
+    it('should approximately generate numbers with similar probability chance', () => {
+      const numbersArr = []
+      const length = 1000
+      const min = 1
+      const max = 5
+
+      for (let index = 0; index < length; index++) {
+        numbersArr.push(pinGenerationService.generateCryptoRandomNumber(min, max))
+      }
+
+      const count = {}
+      for (let num of numbersArr) {
+        count[num] = count[num] ? count[num] + 1 : 1
+      }
+
+      const expectedResult = numbersArr.length / ((max - min) + 1)
+      const errorPercentTolerance = 0.15 // 15%
+      const tolerance = expectedResult * errorPercentTolerance
+
+      for (let [number, freq] of Object.entries(count)) {
+        expect(number >= min && number <= max).toBeTruthy()
+        expect(freq).toBeGreaterThanOrEqual(expectedResult - tolerance)
```

Comment:
this failed on travis already once when tests rerun with my last commit, perhaps the errorPercentTolerance should be changed

Reply:
Upped from 15 - 18%